ddtrace/tracer/tracer.go

```diff
@@ -306,16 +316,50 @@ func (t *tracer) worker(tick <-chan time.Time) {
 	}
 }
 
-func (t *tracer) pushTrace(trace []*span) {
+// traceInfo contains span list and sampling decision of the trace
+// needed to run single span sampling.
+type traceInfo struct {
+	spans            []*span
+	samplingDecision samplingDecision
+}
+
+// processTraceInfo runs single span sampling on the spans pushed to the worker thread.
+// Sampling spans on the worker thread keeps any associated latency off the application thread.
```

Comment:
This also seems out of place and arguably not needed. A function's documentation should not assume that it is called from anywhere. It's just a function at this point.